test/transpile.jl

```diff
@@ -727,9 +783,33 @@ end
     transpiler = CastSwapToCZGateTranspiler()
 
     circuits = [
-        QuantumCircuit(qubit_count = 2, gates = [swap(1, 2)]),
-        QuantumCircuit(qubit_count = 2, gates = [swap(1, 2), x_90(1), swap(1, 2)]),
-        QuantumCircuit(qubit_count = 2, gates = [iswap(1, 2), swap(1, 2)]),
+        QuantumCircuit(qubit_count = 2, instructions = [swap(1, 2)]),
+        QuantumCircuit(qubit_count = 2, instructions = [swap(1, 2), x_90(1), swap(1, 2)]),
+        QuantumCircuit(qubit_count = 2, instructions = [iswap(1, 2), swap(1, 2)]),
+    ]
+
+    for circuit in circuits
+        transpiled_circuit = transpile(transpiler, circuit)
+
+        @test !circuit_contains_gate_type(transpiled_circuit, Snowflurry.Swap)
+        @test compare_circuits(circuit, transpiled_circuit)
+    end
+end
+
+@testset "CastSwapToCZGateTranspiler: Readout" begin
```

Comment:
We have test and make sure that the readouts still make sense here. Implemented by updating `compare_circuits`.

Reply:
not sure if you need me to change something?